### PR TITLE
[JAVA] Remove deprecated Stream-related types

### PR DESCRIPTION
**Pull request by rachel-mack**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-33650>

Staging
 - [Breaking Changes](https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-33650/upgrade/#breaking-changes)
 - [Specify MongoClient Settings](https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-33650/fundamentals/connection/mongoclientsettings/)
 - [Connect to MongoDB by Using a SOCKS5 Proxy](https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-33650/fundamentals/connection/socks/)
 - [Customize TLS/SSL Configuration through the Netty SslContext](https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-33650/fundamentals/connection/tls/#customize-tls-ssl-configuration-through-the-netty-sslcontext)

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/fundamentals/connection/tls.txt [hunk 1]
```diff
@@ -285,9 +282,6 @@ variants the driver supports and the implications of using them.
            .build();
    MongoClientSettings settings = MongoClientSettings.builder()
            .applyToSslSettings(builder -> builder.enabled(true))
-           .streamFactoryFactory(NettyStreamFactoryFactory.builder()
-                   .sslContext(sslContext)
-                   .build())
            .build();
    MongoClient client = MongoClients.create(settings);
```
Comment (rachel-mack) [inline source/fundamentals/connection/tls.txt:286]: For tech review: Can you confirm that this e.g. is still useful.

File: source/fundamentals/connection/tls.txt [hunk 2]
```diff
@@ -272,9 +271,7 @@ provided by Netty.
 
    The driver tests with Netty version ``{+nettyVersion+}``
```
Comment (rachel-mack) [inline source/fundamentals/connection/tls.txt:272]: Tech review: Can you confirm the Netty version number used for testing.

**Review by jordan-smith721 — APPROVED** [review]
LGTM

File: source/fundamentals/connection/tls.txt (hunk 2 above)
Comment (stIncMale) [reply source/fundamentals/connection/tls.txt:272]: It's [4.1.87.Final](https://central.sonatype.com/artifact/io.netty/netty-bom/4.1.87.Final)

File: source/fundamentals/connection/tls.txt [hunk 3]
```diff
@@ -285,9 +282,6 @@ variants the driver supports and the implications of using them.
            .build();
    MongoClientSettings settings = MongoClientSettings.builder()
            .applyToSslSettings(builder -> builder.enabled(true))
-           .streamFactoryFactory(NettyStreamFactoryFactory.builder()
-                   .sslContext(sslContext)
-                   .build())
```
Comment (stIncMale) [inline source/fundamentals/connection/tls.txt:290]: This call should be replaced with something like
```java
.transportSettings(TransportSettings.nettyBuilder()
        .sslContext(sslContext)
        .build())
```

File: source/fundamentals/connection/tls.txt (hunk 1 above)
Comment (stIncMale) [reply source/fundamentals/connection/tls.txt:286]: Yes, it just has to be updated.

File: source/fundamentals/connection/tls.txt [hunk 4]
```diff
@@ -272,9 +271,7 @@ provided by Netty.
 
    The driver tests with Netty version ``{+nettyVersion+}``
 
-To instruct the driver to use `io.netty.handler.ssl.SslContext <https://netty.io/4.1/api/io/netty/handler/ssl/SslContext.html>`__,
-use the `NettyStreamFactoryFactory.Builder.sslContext <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/netty/NettyStreamFactoryFactory.Builder.html#sslContext(io.netty.handler.ssl.SslContext)>`__
-method. See the method documentation for details about the different `io.netty.handler.ssl.SslProvider <https://netty.io/4.1/api/io/netty/handler/ssl/SslProvider.html>`__
```
Comment (stIncMale) [inline source/fundamentals/connection/tls.txt:277]: This information is still needed, but it should point to the methods `com.mongodb.MongoClientSettings.Builder.transportSettings`, `com.mongodb.connection.NettyTransportSettings.Builder.sslContext`. It should also continue pointing to the class `io.netty.handler.ssl.SslProvider`.

The current replacement for this text:

> See the method documentation for details about the different `io.netty.handler.ssl.SslProvider <https://netty.io/4.1/api/io/netty/handler/ssl/SslProvider.html>`__

is weird, as it's unclear what method it refers to.

File: source/upgrade.txt [hunk 5]
```diff
@@ -151,6 +151,24 @@ Version 5.0 Breaking Changes
   responding in the same way as ``Decimal128`` values. In particular,
   ``BsonDecimal128.isNumber()`` now returns ``true``, and
   ``BsonDecimal128.asNumber()`` returns the equivalent ``BsonNumber``.
+  
+- This driver version removes the following methods and types related to the 
+  `StreamFactory
+  <https://mongodb.github.io/mongo-java-driver/4.11/apidocs/mongodb-driver-core/com/mongodb/connection/StreamFactory.html>`__
```
Comment (stIncMale) [inline source/upgrade.txt:157]:
```suggestion
  `Stream
  <https://mongodb.github.io/mongo-java-driver/4.11/apidocs/mongodb-driver-core/com/mongodb/connection/Stream.html>`__
```

**Review by stIncMale — CHANGES_REQUESTED** [review]

**Review by stIncMale — CHANGES_REQUESTED** [review]

File: source/fundamentals/connection/tls.txt [hunk 6]
```diff
@@ -272,25 +271,34 @@ provided by Netty.
...
-variants the driver supports and the implications of using them.
+To instruct the driver to use
+`io.netty.handler.ssl.SslContext <https://netty.io/4.1/api/io/netty/handler/ssl/SslProvider.html>`__,
```
Comment (stIncMale) [inline source/fundamentals/connection/tls.txt:275]: This looks like a typo. The link should be to `SslContext`, not `SslProvider`.

**Review by stIncMale — CHANGES_REQUESTED** [review]

**Review by stIncMale — APPROVED** [review]

**Pull request merged**